modules/o2e_obe_salesforce/src/ObeSfLogger.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Drupal\o2e_obe_salesforce;
+
+use Drupal\Core\Logger\LoggerChannelFactory;
+
+/**
+ * Defines a Class for logging Obe Sf Logger.
+ */
+class ObeSfLogger {
+
+  /**
+   * Logger Factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactory
+   */
+  protected $loggerFactory;
+
+  /**
+   * Constructor method.
+   */
+  public function __construct(LoggerChannelFactory $logger_factory) {
+    $this->loggerFactory = $logger_factory;
+  }
+
+  /**
+   *
+   */
+  public function log(string $channel = NULL, string $error_type = NULL, string $error_msg = NULL, string $endpoint = NULL, string $endpointType = NULL, string $payload = NULL, string $result = NULL) {
```

Comment:
```suggestion
  public function log(string $channel, string $severity, string $message, array $context = [], array $request_info = []) {
```
1. Channel can never be empty.
2. $severity -> log message severity i.e. error, critical, notice, etc.
3. $message -> Actual error or message to be logged with placeholders.
4. $context -> array of placeholder values to be replaced in $message.
5. $request_info -> Can contain the following
    1. request_url -> endpoint URL used for guzzle request
    2. type -> GET or POST
    3. payload -> values passed to the request
    4. response -> response from the API.